packages/wallet/src/services/TransactionReemitter.ts

```diff
@@ -139,9 +141,9 @@ export const createTransactionReemitter = ({
     mergeMap((txs) => from(txs))
   );
 
-  const reemitUnconfirmed$ = combineLatest([tipSlot$, inFlight$]).pipe(
-    mergeMap(([tip, inFlight]) => {
-      const reemitSubmittedBefore = tip - reemitUnconfirmedAfterSlots;
+  const reemitUnconfirmed$ = combineLatest([tipSlot$, inFlight$, genesisParameters$]).pipe(
```

Comment:
There might be a corner case when genesisParameters$ emits twice (epoch change?).
combineLatest will make reemitUnconfirmed$ to emit the same inFlight transactions again.
```
tipSlot$ =  '         a-'
inFlight$ = '         b-'
genesisParameters$ = 'x-y--'
combineLatest  = '    (abx)-(aby)'
reemitUnconfirmed$ = 'b-b'; // same transaction emitted twice

Reply:
Nice catch! This has actually crossed my mind too. An alternative has a caveat too:
```
tipSlot$ =  '         a-c'
genesisParameters$ = '-b'
reemitUnconfirmed$ = '--c';
// if genesisParameters$ is not available yet when tip$ is emitted,
// it will get delayed until another emission of tip$
```

8a4c691d6520bbd7aca6de9e6827af4857c585bb I think that's slightly better than original implementation